### ISSUE TITLE
can you add the group normalization in your code?

### ISSUE THREAD

**Issue by bubulv** [pjreddie/darknet#1119]
In your cfg , if  batch=32 and subdivisions=8, so the real batch_size is 32/8=4. Does BN operate on  4 images or 32 images? 

Hope you can publish the GN code!!! Thank you very much

**Comment by drsxr** [comment]
Good question, as we know that BatchNorm starts failing with N<16 and gets inaccurate down towards N<=4.  GroupNorm or InstanceNorm better alternatives in that case.

**Comment by arnaud-nt2i** [comment]
up !

**Comment by arnaud-nt2i** [comment]
@bubulv @drsxr 
Hi ! according to AlexeyAB on [https://github.com/AlexeyAB/darknet/issues/4386](url):

`
I implemented part of CBN - averaging statistic inside one batch.
So you can increase accuracy just by increasing batch= in cfg-file, and set cbn=1 instead of batch_normalize=1
So batch=120 subdivisions=4 with CBN, should work better than batch=120 subdivisions=4 with BN.
But batch=120 subdivisions=4 with CBN, will work worse than batch=120 subdivisions=1 with BN.
`

`
I.e. using batch=64 subdivisions=8 with BN, avg mini_batch_size = 8
64/8 = 8
`

`
I.e. using batch=64 subdivisions=8 with CBN, avg mini_batch_size = 36
(8+16+24+32+40+48+56+64)/8 = 36
`

`
You can try it on Classifier csresnext50
`

`
So inside 1 batch it will average the values of Mean and Variance.
I.e if you train with batch=64 subdivisions=16, then will be 16 mini_batches with size 4.
`

`
For the 1st mini_batch will use Mean[1] & Variance[1]
For the 2nd mini_batch will use avg(Mean[1], Mean[2]) & avg(Variance[1], Variance[2])
For the 3rd mini_batch will use avg(Mean[1], Mean[2], Mean[3]) & avg(Variance[1], Variance[2], Variance[3])
`

This mean BN operate on mini-batch size (32/8=4 on your example)
CBN (only on AlexeyAB repo) on the contrary operates on successive mini-batch computation.
